substrate/frame/support/src/lib.rs

```diff
@@ -1562,6 +1562,57 @@ pub mod pallet_macros {
 	/// * [`frame_support::derive_impl`].
 	/// * [`#[pallet::no_default]`](`no_default`)
 	/// * [`#[pallet::no_default_bounds]`](`no_default_bounds`)
+	///
+	/// ## Optional: `without_metadata`
+	///
+	/// By default, the associated types of the `Config` trait that require the `TypeInfo` or
+	/// `Parameter` bounds are included in the metadata of the pallet. Please note that the
+	/// `RuntimeEvent` and constants are already included in the metadata.
+	///
+	/// The optional `without_metadata` argument can be used to exclude these associated types
+	/// from the metadata collection.
+	///
+	/// Furthermore, the `without_metadata` argument can be used in combination with the
+	/// [`#[pallet::include_metadata]`](`include_metadata`) attribute to selectively include
+	/// only certain associated types in the metadata collection.
+	///
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	/// # 	use frame_support::pallet_prelude::*;
+	/// # 	use frame_system::pallet_prelude::*;
+	/// # 	use core::fmt::Debug;
+	/// # 	use frame_support::traits::Contains;
+	/// #
+	/// # 	pub trait SomeMoreComplexBound {}
+	/// #
+	/// 	#[pallet::pallet]
+	/// 	pub struct Pallet<T>(_);
+	///
+	/// 	#[pallet::config(with_default, without_metadata)] // <- with_default and without_metadata are optional
+	/// 	pub trait Config: frame_system::Config {
+	/// 		/// The overarching event type.
+	/// 		#[pallet::no_default_bounds] // Default with bounds is not supported for RuntimeEvent
+	/// 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+	///
+	/// 		/// A simple type.
+	/// 		// Type that would have been included in metadata, but is now excluded.
+	/// 		type SimpleType: From<u32> + TypeInfo;
+	///
+	/// 		// The `pallet::include_metadata` is used to selectively include this type in metadata.
+	/// 		#[pallet::include_metadata]
+	/// 		type SelectivelyInclude: From<u32> + TypeInfo;
+	/// 	}
+	///
+	/// 	#[pallet::event]
+	/// 	pub enum Event<T: Config> {
+	/// 		SomeEvent(u16, u32),
+	/// 	}
+	/// }
+	/// ```
+	///
+	/// For more information, see:
+	/// * [`#[pallet::include_metadata]`](`include_metadata`)
```

Comment:
Why are you linking `include_metadata` when talking about `without_metadata`?

Reply:
This other macro does not really provides more information. More like gives you the opportunity to include selected items.